.github/workflows/build-ubuntu.yaml

```diff
@@ -0,0 +1,59 @@
+name: build-ubuntu
+
+
+# Controls when the action will run
+on:
+  push:
+  pull_request:
+  workflow_dispatch: # trigger the workflow manually
+  # # trigger the workflow on schedule
+  # # https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#scheduled-events
+  # schedule:
+  #   - cron:  '* * * * *'
+
+
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build:
+    runs-on: ${{ matrix.runner }}
+    continue-on-error: false
+    strategy:
+      matrix:
+        runner: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
+        compiler: [clang++, g++]
+        python: [python2, python3]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install requirements
+        run: >
+          sudo apt install --no-install-recommends -y
+          build-essential
```

Comment:
Interestingly `build-essentials` doesn't bring in `g++`, just `gcc`. I think `clang` brings in `g++` as a dependence, but don't be surprised if it doesn't and you see errors.